qemu/tests/boot_from_device.py

```diff
@@ -88,15 +87,17 @@ def check_boot_result(boot_fail_info, device_name):
             except ValueError:
                 output = vm.serial_console.get_output()
             match = re.search(infos[0], output)
-            if match or time.time() > start + timeout:
+            if match:
                 break
+            if time.time() > start + timeout:
+                test.fail("No device for boot")
```

Comment:
I think this should be a test.error.
It might be better to log how long have been past eg `''No device for boot after %s" % timeout`
or "`Can not boot up after %s" % timeout`

Reply:
`test.fail` should be more accurate, but yes, the error message should be improved.